### PR TITLE
Added Post Endpoint for single record in students table

### LINKED ISSUE #35: Add Post endpoint for single record in students table
See #26

### PR DESCRIPTION
**Pull request by smmzhu**
## Overview

(Merge https://github.com/ucsb-cs156-f24/proj-happycows-f24-11/pull/30 before this one)
Third ticket in a series tackling issue https://github.com/ucsb-cs156-f24/proj-happycows-f24-11/issues/26 — added Post Endpoint for single record in students table.
Testing Link: https://sampr2.dokku-11.cs.ucsb.edu/

## Screenshots (Optional)

image

## Validation (Optional)

1. Open swagger.
2. Make sure the post endpoint is visible.
3. Test with an example query that the endpoint works as intended.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [ ] ~~Frontend Unit tests (`npm test`) pass~~
- [ ] ~~Frontend Test coverage (`npm run coverage`) 100%~~
- [ ] ~~Frontend Mutation tests (`npx stryker run`) 100%~~ 
- [ ] ~~Frontend Linting (`npx eslint --fix src`)~~ 

## Linked Issues

Closes #35

### CODE REVIEW

**Review by adenjo6 — DISMISSED** [review]
lgtm

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @PostMapping("")
+    public Student postStudent(
+        @Parameter(name = "courseId") @RequestParam Long courseId,
+        @Parameter(name = "fname") @RequestParam String fname,
+        @Parameter(name = "lname") @RequestParam String lname,
+        @Parameter(name = "studentId") @RequestParam String studentId,
+        @Parameter(name = "email") @RequestParam String email
+    ) {
+        Student student = new Student();
+        student.setCourseId(courseId);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentController.java:50]: You should verify that the courseId exists here and return an error if it does not.  See for example, lines 114-118 here:

https://github.com/ucsb-cs156-f24/proj-happycows-f24-11/blob/73f2da95081d339fe8c74ceab1ff36a8e5c620f1/src/main/java/edu/ucsb/cs156/happiercows/controllers/ChatMessageController.java#L114

**Review by pconrad — CHANGES_REQUESTED** [review]
One small change needed: validate the courseId when creating a student.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentController.java (hunk 1 above)
Comment (smmzhu) [reply src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentController.java:58]: fixed!

**Review by blobthebuilder — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**